cmd/dist/fetch.go

```diff
@@ -0,0 +1,254 @@
+package main
+
+import (
+	contextpkg "context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/containerd/log"
+	"github.com/docker/containerd/remotes"
+	"github.com/docker/distribution/manifest/schema2"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+var fetchCommand = cli.Command{
+	Name:        "fetch",
+	Usage:       "retrieve objects from a remote",
+	ArgsUsage:   "[flags] <remote> <object> [<hint>, ...]",
+	Description: `Fetch objects by identifier from a remote.`,
+	Flags: []cli.Flag{
+		cli.DurationFlag{
+			Name:   "timeout",
+			Usage:  "total timeout for fetch",
+			EnvVar: "CONTAINERD_FETCH_TIMEOUT",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		ctx := contextpkg.Background()
+
+		timeout := context.Duration("timeout")
+		if timeout > 0 {
+			var cancel func()
+			ctx, cancel = contextpkg.WithTimeout(ctx, timeout)
+			defer cancel()
+		}
+
+		var (
```

Comment:
Provides a nice pre-declaration block. This is a little sloppy: `timeout` and `context` should be here to.